### PR TITLE
Document access to Azure Service Hubs and Red Hat OpenShift Streams for Apache Kafka

### LINKED ISSUE #24726: Document how to use Azure Service Hub (using the Kafka protocol)
### Description

Add a section in the Kafka reference guide that covers configuring the application to use the Azure Service hub.

### Implementation ideas

Here is what should be added:

```properties
kafka.bootstrap.servers=my-event-hub.servicebus.windows.net:9093
kafka.security.protocol=SASL_SSL
kafka.sasl.mechanism=PLAIN
kafka.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
    username="$ConnectionString" \
    password="Endpoint=sb://my-event-hub.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=...";
```

@agoncal Is `$ConnectionString` really `$ConnectionString` or a substituted variable?

### PR DESCRIPTION
**Pull request by cescoffier**
Fix https://github.com/quarkusio/quarkus/issues/24726

### CODE REVIEW

File: docs/src/main/asciidoc/kafka.adoc [hunk 1]
```diff
@@ -2401,6 +2401,95 @@ quarkus.log.category."org.apache.kafka.common.utils".level=INFO
...
+kafka.security.protocol=SASL_SSL
+kafka.sasl.mechanism=PLAIN
+kafka.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \ # <2>
+    username="$ConnectionString" \ # <3>
+    password="<YOUR.EVENTHUBS.CONNECTION.STRING>"; # <4>
+----
+1. The port is `9093`.
+2. You need to use the JAAS `PlainLoginModule`.
+3. The username is the `$ConnectionString` string.
+4. The Event Hub connection string given by Azure.
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka.adoc:2425]:
```suggestion
<1> The port is `9093`.
<2> You need to use the JAAS `PlainLoginModule`.
<3> The username is the `$ConnectionString` string.
<4> The Event Hub connection string given by Azure.
```

File: docs/src/main/asciidoc/kafka.adoc [hunk 2]
```diff
@@ -2401,6 +2401,95 @@ quarkus.log.category."org.apache.kafka.common.utils".level=INFO
...
+kafka.bootstrap.servers=<connection url> # <1>
+kafka.security.protocol=SASL_SSL
+kafka.sasl.mechanism=PLAIN
+kafka.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
+  username="${KAFKA_USERNAME}" \ # <2>
+  password="${KAFKA_PASSWORD}"; # <3>
+----
+1. The connection string, given on the admin console, such as `demo-c--bjsv-ldd-cvavkc-a.bf2.kafka.rhcloud.com:443`
+2. The kafka username (the client id from the service account)
+3. the kafka password (the client secret from the service account)
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka.adoc:2466]:
```suggestion
<1> The connection string, given on the admin console, such as `demo-c--bjsv-ldd-cvavkc-a.bf2.kafka.rhcloud.com:443`
<2> The Kafka username (the client id from the service account)
<3> the Kafka password (the client secret from the service account)
```

File: docs/src/main/asciidoc/kafka.adoc [hunk 3]
```diff
@@ -2401,6 +2401,95 @@ quarkus.log.category."org.apache.kafka.common.utils".level=INFO
...
+
+NOTE: In general, these properties are prefixed using `%prod` to enable them only when running in production mode.
+
+IMPORTANT: To use Red Hat OpenShift Streams for Apache Kafka, you need to create the topic beforehand, create a _Service Account_,
+and provide permissions to read and write to your topic from that service account. The authentication data related to the service account.
```
Comment (gsmet) [inline docs/src/main/asciidoc/kafka.adoc:2471]: > The authentication data related to the service account.

This sentence is incomplete AFAICS.

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a few minor issues.

File: docs/src/main/asciidoc/kafka.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/kafka.adoc:2425]: Formatted as you did, it's just a list not pointing to the callouts.

**Comment by gsmet** [comment]
Not sure if it's working with 2.8 already but if so, let's mark it for backport please.

File: docs/src/main/asciidoc/kafka.adoc [hunk 4]
```diff
@@ -2401,6 +2401,95 @@ quarkus.log.category."org.apache.kafka.common.utils".level=INFO
...
+  username="${KAFKA_USERNAME}" \ # <2>
+  password="${KAFKA_PASSWORD}"; # <3>
+----
+<1> The connection string, given on the admin console, such as `demo-c--bjsv-ldd-cvavkc-a.bf2.kafka.rhcloud.com:443`
+<2> The kafka username (the client id from the service account)
+<3> the kafka password (the client secret from the service account)
+
+NOTE: In general, these properties are prefixed using `%prod` to enable them only when running in production mode.
+
+IMPORTANT: To use Red Hat OpenShift Streams for Apache Kafka, you must create the topic beforehand, create a _Service Account_, and provide permissions to read and write to your topic from that service account.
```
Comment (ozangunalp) [inline docs/src/main/asciidoc/kafka.adoc:2470]: Would be useful to add a link to this guide for these operations:
 https://access.redhat.com/documentation/en-us/red_hat_openshift_streams_for_apache_kafka/1/guide/88e1487a-2a14-4b35-85b9-a7a2d67a37f3

**Review by ozangunalp — CHANGES_REQUESTED** [review]
Looks good to me!

File: docs/src/main/asciidoc/kafka.adoc (hunk 4 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/kafka.adoc:2470]: How stable is this link? My bookmark are regularly out of date.

Comment (ozangunalp) [reply docs/src/main/asciidoc/kafka.adoc:2470]: Yeah, I agree. I have some 404 pages through the web archive.

Comment (edeandrea) [reply docs/src/main/asciidoc/kafka.adoc:2470]: @cescoffier I have a blog post that specifically shows how to do this using the Quarkus Superheroes sample app - https://developers.redhat.com/articles/2022/03/23/quarkus-superheroes-managed-services-save-day

Feel free to link to it if you feel its appropriate.

Comment (cescoffier) [reply docs/src/main/asciidoc/kafka.adoc:2470]: ah yes, good idea!

Comment (cescoffier) [reply docs/src/main/asciidoc/kafka.adoc:2470]: After verification, the https://access.redhat.com/documentation/en-us/red_hat_openshift_streams_for_apache_kafka/1/guide/88e1487a-2a14-4b35-85b9-a7a2d67a37f3 should be stable (that's what took me some time to come back to this PR).

**Review by gsmet — APPROVED** [review]

**Comment by edeandrea** [comment]
@cescoffier what about documenting how to use the schema registry in azure event hub? Or is that not possible with Quarkus?

**Comment by cescoffier** [comment]
Schema Registry will be documented in a second step.

Regarding Azure, without an Azure account, it's not possible for us to test it. So, until we have more time (and budget) we cannot really test it.

**Comment by cescoffier** [comment]
@edeandrea I would say it's possible with the right config. Now using the Azure serde is likely not going to work in native without some substitutions.

**Pull request merged**